src/s_sound.c

```diff
@@ -710,11 +710,6 @@ void S_ChangeMusInfoMusic (int lumpnum, int looping)
 
    music = &S_music[mus_musinfo];
 
-   if (music->lumpnum == lumpnum)
```

Comment:
Could we somehow keep this? I am afraid this will cause some music to restart when it should just continue.

Reply:
I don't know, I think we can risk it. Tried various scenarios - it seems to work correctly. If the music restarts, it's not the end of the world. If someone finds such case, we can fix it.

Reply:
I think this can happen if we start a map with normal music and then walk into a sector with a music changer thing which sets the music to `0`, i.e. the map's default music.

Reply:
Interesting, I'll try to make a test map later.

Reply:
Test WAD: [MusChange.zip](https://github.com/fabiangreffrath/woof/files/10148633/MusChange.zip)
I tried different MUSINFO, but it seems to work, I'm not sure if I did it right.

Reply:
> Test WAD: [MusChange.zip](https://github.com/fabiangreffrath/woof/files/10148633/MusChange.zip) I tried different MUSINFO, but it seems to work, I'm not sure if I did it right.

The map should have three sectors: The player starts in a sector without a music changer thing. He should be able to either move into a sector with a music changer thing type `14100`, which does not require an entry in the MUSINFO lump and just falls back to the map's default music. And the player should be able to move into a sector with a music changer thing type `14101` with a corresponding entry in the MUSINFO lump pointing it to D_RUNNIN.

In both sectors, the music shouldn't start anew.

Reply:
I've made 3 sectors MAP, seems to work: [MusChange2.zip](https://github.com/fabiangreffrath/woof/files/10154400/MusChange2.zip)

Reply:
> Could you check this map? Sorry, I cannot currently test game music...

It works with this MUSINFO
```
MAP01
1 JJ2HELL
```
It's almost exactly MusChange2.zip 😄

Reply:
Well, the point was to check whether entering a sector with a music changer thing that triggers the exact same music that is already running will cause the music to restart. Thus I left out the music changer for JJ2HELL.

Reply:
With the original MUSINFO also does not restart. I think it works correctly.

Reply:
> So I think it works. Should we merge this?

Yes, just checked, it works!